libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -264,8 +264,9 @@ bool AP_Arming::barometer_checks(bool report)
 #endif
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
-            check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
+        char buffer[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1] {};
+        if (!AP::baro().arming_checks(sizeof(buffer), buffer)) {
+            check_failed(ARMING_CHECK_BARO, report, "%s", buffer);
```

Comment:
```suggestion
            check_failed(ARMING_CHECK_BARO, report, "Baro: %s", buffer);
```

we've used this pattern for other libraries and it seems to work well - you can see that pattern used for many libraries called in this file.

Reply:
changed